### PR TITLE
issue #563 fixed changed to upper case

### LINKED ISSUE #563: Battleship Cog Causes IndexError
## Description

The `battleship` cog runs into an error when given lowercased input. An IndexError is caused when trying to get the matching letter from the list of letters. The bot quits the game with the error:
```
An error occurred. Game failed
```

## Steps to Reproduce

1. Start a game of battleship
2. Send a DM containing any spot in lowercase on your turn (example: `a1`)
3. The bot sends the failure message in the commands channel, the index error is raised to console

## Expected Behaviour

The bot should be able to convert between upper and lower case.

## Actual Behaviour

The bot errors out on lower case. Error:
```py
File "./sir-lancebot/bot/exts/evergreen/battleship.py", line 146, in get_square
     return grid[number-1][index]  # -1 since lists are indexed from 0
IndexError: list index out of range
```

## Possible Solutions

You could try converting from lower to uppercase.

Additionally, you can add a try/except around the input section, to report invalid input and stop the bot from failing the game.

## Would you like to implement a fix?

***Note: For high-priority or critical bugs, fixes may be implemented by staff.***

- [ ] I'd like to implement the bug fix
- [x] Anyone can implement the bug fix

### PR DESCRIPTION
**Pull request by rijusougata13**
## Relevant Issues
"Closes #563 "

## Description
 changed index to upper so it will not out of bound 

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x ] If dependencies have been added or updated, run `pipenv lock`?
- [x ] **Lint your code** (`pipenv run lint`)?
- [ x] Set the PR to **allow edits from contributors**?

### CODE REVIEW

File: bot/exts/evergreen/battleship.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def get_square(grid: Grid, square: str) -> Square:
         """Grabs a square from a grid with an inputted key."""
         index = ord(square[0]) - ord("A")
         number = int(square[1:])
-
+        index.upper()
```
Comment (HassanAbouelela) [inline bot/exts/evergreen/battleship.py:145]: Index here is a number, you need to uppercase the letter before `ord()` gets called on it in line 143.

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]

**Comment by HassanAbouelela** [comment]
Hey, I appreciate the enthusiasm, but could you check in on the [dev-contrib discord channel](https://discord.com/channels/267624335836053506/635950537262759947/801538814215585802)? I'd just like to discuss somethings.

In terms of the PR, the proper procedure would be to leave a comment on the relevant issue so you can be assigned, and to fill in the PR form here.

File: bot/exts/evergreen/battleship.py [hunk 2]
```diff
@@ -140,9 +140,11 @@ def format_grid(player: Player, emojiset: EmojiSet) -> str:
     @staticmethod
     def get_square(grid: Grid, square: str) -> Square:
         """Grabs a square from a grid with an inputted key."""
+        #changed to upper case
+        square[0].upper()
```
Comment (HassanAbouelela) [inline bot/exts/evergreen/battleship.py:144]: String.upper is non-mutatating, meaning this change doesn't actually change the value stored in the list. You have to store this value and pass it to the function on the next line, or call upper on the next line directly.

**Review by HassanAbouelela — CHANGES_REQUESTED** [review]

File: bot/exts/evergreen/battleship.py (hunk 2 above)
Comment (rijusougata13) [reply bot/exts/evergreen/battleship.py]: ind=square[0].upper()
index = ord(ind) - ord("A") 
is this ok?

File: bot/exts/evergreen/battleship.py [hunk 3]
```diff
@@ -140,9 +140,10 @@ def format_grid(player: Player, emojiset: EmojiSet) -> str:
     @staticmethod
     def get_square(grid: Grid, square: str) -> Square:
         """Grabs a square from a grid with an inputted key."""
-        index = ord(square[0]) - ord("A")
+        #changed to upper case
+        ind=square[0].upper()
```
Comment (Akarys42) [inline bot/exts/evergreen/battleship.py:144]: No needs for an additional variable, this can be done in-line with the next line.

File: bot/exts/evergreen/battleship.py [hunk 4]
```diff
@@ -140,9 +140,10 @@ def format_grid(player: Player, emojiset: EmojiSet) -> str:
     @staticmethod
     def get_square(grid: Grid, square: str) -> Square:
         """Grabs a square from a grid with an inputted key."""
-        index = ord(square[0]) - ord("A")
+        #changed to upper case
```
Comment (Akarys42) [inline bot/exts/evergreen/battleship.py:143]: Please conform to the PEP8, two spaces after the hashtag.

File: bot/exts/evergreen/battleship.py [hunk 5]
```diff
@@ -140,9 +140,10 @@ def format_grid(player: Player, emojiset: EmojiSet) -> str:
...
+        ind=square[0].upper()
+        index = ord(ind) - ord("A")
         number = int(square[1:])
-
```
Comment (Akarys42) [inline bot/exts/evergreen/battleship.py:145]: Please revert this change

**Comment by rijusougata13** [comment]
> Hey, I appreciate the enthusiasm, but could you check in on the [dev-contrib discord channel](https://discord.com/channels/267624335836053506/635950537262759947/801538814215585802)? I'd just like to discuss some things.
> 
> In terms of the PR, the proper procedure would be to leave a comment on the relevant issue so you can be assigned, and to fill in the PR form here.

can you explain why my PR is not accepting should i have to assign myself first?

**Review by Akarys42 — CHANGES_REQUESTED** [review]
Hello, and thanks for your contribution!

There are some issues with your PR, can you please fix those? 

I'd also like you to try to write more meaningful commit messages. When reading through the history, just having `fixed` isn't very helpful.

By the way, what is your Discord username?

File: bot/exts/evergreen/battleship.py (hunk 3 above)
Comment (rijusougata13) [reply bot/exts/evergreen/battleship.py:144]: here is my discord username :𝖆𝖒𝖆𝖙𝖊𝖗𝖆𝖘𝖚#6340  ok i will write a meaningful commit message and can you help me out with this.I am new in this open source .but i cant understand whats going on

**Review by Akarys42 — APPROVED** [review]

**Review by ks129 — APPROVED** [review]

**Review by Xithrius — APPROVED** [review]

**Review by HassanAbouelela — APPROVED** [review]
